sdk/tables/data-tables/src/TableBatch.ts

```diff
@@ -0,0 +1,294 @@
+    if (!credential) {
+      // When authenticating with SAS we need to add the SAS token after $batch
+      const urlParts = url.split("?");
+      this.url = urlParts[0];
+      const sas = urlParts.length > 1 ? `?${urlParts[1]}` : "";
+      this.url = `${this.url}/$batch${sas}`;
+    } else {
+      // When using a SharedKey credential no SAS token is needed
+      this.url = `${this.url}/$batch`;
+    }
+  }
+
+  /**
+   * Adds a createEntity operation to the batch
+   * @param entity Entity to create
+   */
+  public createEntity<T extends object>(entity: TableEntity<T>): void {
+    this.checkPartitionKey(entity.partitionKey);
+    this.pendingOperations.push(this.interceptClient.createEntity(entity));
+  }
+
+  /**
+   * Adds a createEntity operation to the batch per each entity in the entities array
+   * @param entitites Array of entities to create
+   */
+  public createEntities<T extends object>(entitites: TableEntity<T>[]): void {
+    for (const entity of entitites) {
+      this.checkPartitionKey(entity.partitionKey);
+      this.pendingOperations.push(this.interceptClient.createEntity(entity));
+    }
+  }
+
+  /**
+   * Adds a deleteEntity operation to the batch
+   * @param partitionKey partition key of the entity to delete
+   * @param rowKey row key of the entity to delete
+   * @param options options for the delete operation
+   */
+  public deleteEntity(
+    partitionKey: string,
+    rowKey: string,
+    options?: DeleteTableEntityOptions
+  ): void {
+    this.checkPartitionKey(partitionKey);
+    this.pendingOperations.push(this.interceptClient.deleteEntity(partitionKey, rowKey, options));
+  }
+
+  /**
+   * Adds an updateEntity operation to the batch
+   * @param entity entity to update
+   * @param mode update mode (Merge or Replace)
+   * @param options options for the update operation
+   */
+  public updateEntity<T extends object>(
+    entity: TableEntity<T>,
+    mode: UpdateMode,
+    options?: UpdateTableEntityOptions
+  ): void {
+    this.checkPartitionKey(entity.partitionKey);
+    this.pendingOperations.push(this.interceptClient.updateEntity(entity, mode, options));
+  }
+
+  /**
+   * Submits the operations in the batch
+   */
+  public async submitBatch(): Promise<any> {
+    await Promise.all(this.pendingOperations);
+    const body = this.batchRequest.getHttpRequestBody();
+    const client = new ServiceClient();
+    const headers: RawHttpHeaders = {
+      accept: "application/json",
+      "x-ms-version": "2019-02-02",
+      "Accept-Charset": "UTF-8",
+      DataServiceVersion: "3.0;",
+      MaxDataServiceVersion: "3.0;NetFx",
+      "Content-Type": `multipart/mixed; boundary=batch_${this.batchGuid}`,
+      Connection: "Keep-Alive"
+    };
+
+    const { span, updatedOptions } = createSpan("TableBatch-submitBatch", {});
+    const request = new WebResource(this.url, "POST", body, undefined, new HttpHeaders(headers));
+
+    request.spanOptions = updatedOptions;
```

Comment:
`createSpan` returns `OperationOptions`, right? So shouldn't this be

```suggestion
    request.spanOptions = updatedOptions.spanOptions;
```

I'm kinda surprised that `{}` is allowed to satisfy `extends OperationOptions`, but I suppose it's because it's entirely optional.

Reply:
oh! good catch thanks!